sklearn/neighbors/_classification.py

```diff
@@ -199,6 +199,15 @@ def fit(self, X, y):
 
         return self._fit(X, y)
 
+    def _build_sparse_matrix(self, neighbor_labels, weights):
+        data = weights.ravel()
+        indices = neighbor_labels.ravel()
+        indptr = np.arange(neighbor_labels.shape[0] + 1) * self.n_neighbors
+        return sparse.csr_matrix(
+            (data, indices, indptr),
+            shape=(neighbor_labels.shape[0], neighbor_labels.max() + 1),
+        )
```

Comment:
Let's rename `neighbor_labels` to `neighbor_class_idx` as the mapping from a class index (contiguous int values starting at 0 for each output k) to a class label (any int or str object) will be done afterwards using `self.classes_`.

Furthermore we can drop the weights argument since it will always be ones in this code. This makes the code easier to understand (I think).

```suggestion
    def _sparse_class_counts(self, neighbors_class_indices):
        """Sparse class count encoding of neighbors classes
        
        Convert a dense numpy array of class integer indices for the results of
        neighbors queries into a sparse CSR matrix with class counts.
        
        The sparse.csr_matrix constructor automatically sums repeated count
        values in case a given query has several neighbors of the same class. 
        """
        n_queries, n_neighbors = neighbors_class_indices.shape
        data = np.ones(shape=n_queries * n_neighbors, dtype=np.uint32)
        indices = neighbors_class_indices.ravel()
        indptr = np.arange(n_queries + 1) * n_neighbors
        return sparse.csr_matrix(
            (data, indices, indptr),
            shape=(n_queries, neighbors_class_indices.max() + 1),
        )
```

Not that I used `np.uint32` for the dtype of the class counts, because:

- we will never have negative counts by construction;
- we will always have n_neighbors << np.info(np.uint32).max, so there is no need to use a 64 bit integer here.

Reply:
Also we could move this into a module level (private) function instead of a method because it does not need `self`.

Please also write some unit tests dedicated on some toy data.